src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -424,6 +426,8 @@ protected internal virtual void OnProjectRemoved(ProjectId projectId)
                 var newSolution = this.SetCurrentSolution(oldSolution.RemoveProject(projectId));
 
                 this.RaiseWorkspaceChangedEventAsync(WorkspaceChangeKind.ProjectRemoved, oldSolution, newSolution, projectId);
+
+                _isProjectUnloading = false;
```

Comment:
Is anything ever expected to throw between these two points?  Should this be in a `finally`?